### PR TITLE
Change boto3 version to 1.28.64 for 1.0.0 release

### PR DESCRIPTION
**Pull request by zuoyuanh**
*Description of changes:*

Change boto3 version to 1.28.64 for 1.0.0 release

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by just4brown — APPROVED** [review]

**Review by ketan-vijayvargiya — DISMISSED** [review]

File: build_artifacts/v1/v1.0/v1.0.0/gpu.env.in [hunk 1]
```diff
@@ -17,7 +17,7 @@ conda-forge::py-xgboost-gpu[version='>=1.7.6']
 conda-forge::jupyterlab[version='>=3.6.5']
 conda-forge::ipywidgets[version='>=8.1.0']
 conda-forge::conda[version='>=23.7.3']
-conda-forge::boto3[version='>=1.28.35']
+conda-forge::boto3[version='>=1.28.35,<1.28.65']
```
Comment (just4brown) [inline build_artifacts/v1/v1.0/v1.0.0/gpu.env.in:20]: Will need identical changes to this file as compared with CPU. The `botocore` constraint is missing here

Comment (zuoyuanh) [reply build_artifacts/v1/v1.0/v1.0.0/gpu.env.in:20]: Thanks for the catch - fixed it for GPU

**Review by just4brown — APPROVED** [review]

**Comment by balajisankar15** [comment]
@zuoyuanh Can you close this if this is no longer relevant?

**Comment by claytonparnell** [comment]
No longer relevant

**Pull request closed without merging**